### ISSUE TITLE
reportIncompatibleMethodOverride when subclass uses a property defined with setter

### ISSUE THREAD

**Issue by odrling** [microsoft/pyright#7478]
With pyright 1.1.354, properties defined with a setter are not allowed to be used in a subclass (reportIncompatibleMethodOverride)

Can be reproduced with the following code:
```python
class PropertyBase:

    def __init__(self):
        self._a = True

    @property
    def a(self):
        return self._a

    @a.setter
    def a(self, value: bool):
        self._a = value


class Property(PropertyBase):

    def __init__(self):
        self.a = False
```

```
/tmp/test.py
  /tmp/test.py:18:14 - error: "a" incorrectly overrides property of same name in class "PropertyBase" (reportIncompatibleMethodOverride)
1 error, 0 warnings, 0 informations 
```

This happens since 3ccd7b87b5a4c23a693d049c402bad63a5462bd7

**Comment by erictraut** [comment]
Thanks for reporting. Apologies for the regression. I'm surprised that our unit tests and regression tests didn't catch this. I'll work on a fix to both the underlying cause of the bug and the hole in our test suite.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.355, which I just published.
